From cf723cfb6d24463c61b23ffa1399dcb8fc7037c0 Mon Sep 17 00:00:00 2001
From: Stephan Mueller <smueller@chronox.de>
Date: Thu, 12 Jan 2023 21:26:41 +0100
Subject: [PATCH 34/35] lrng: forced reseeding if fast entropy sources are
 available

Replace lrng_es_add_entropy with lrng_force_fully_seeded for all occasions
where a fast entropy source is possibly present.

Signed-off-by: Stephan Mueller <smueller@chronox.de>
---
 drivers/char/lrng/lrng_es_krng.c | 2 +-
 drivers/char/lrng/lrng_es_mgr.c  | 2 +-
 drivers/char/lrng/lrng_switch.c  | 4 ++--
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/char/lrng/lrng_es_krng.c b/drivers/char/lrng/lrng_es_krng.c
index 519ba640c..8844e8133 100644
--- a/drivers/char/lrng/lrng_es_krng.c
+++ b/drivers/char/lrng/lrng_es_krng.c
@@ -39,7 +39,7 @@ static int lrng_krng_adjust_entropy(void)
 		 entropylevel);
 	lrng_drng_force_reseed();
 	if (entropylevel)
-		lrng_es_add_entropy();
+		lrng_force_fully_seeded();
 	return 0;
 }
 
diff --git a/drivers/char/lrng/lrng_es_mgr.c b/drivers/char/lrng/lrng_es_mgr.c
index c74108c1e..1720df757 100644
--- a/drivers/char/lrng/lrng_es_mgr.c
+++ b/drivers/char/lrng/lrng_es_mgr.c
@@ -262,7 +262,7 @@ void lrng_unset_fully_seeded(struct lrng_drng *drng)
 		lrng_state.lrng_fully_seeded = false;
 
 		/* If sufficient entropy is available, reseed now. */
-		lrng_es_add_entropy();
+		lrng_force_fully_seeded();
 	}
 }
 
diff --git a/drivers/char/lrng/lrng_switch.c b/drivers/char/lrng/lrng_switch.c
index aae755949..2df723489 100644
--- a/drivers/char/lrng/lrng_switch.c
+++ b/drivers/char/lrng/lrng_switch.c
@@ -227,7 +227,7 @@ int lrng_set_drng_cb(const struct lrng_drng_cb *drng_cb)
 	ret = lrng_switch(drng_cb, lrng_drng_switch);
 	/* The switch may imply new entropy due to larger DRNG sec strength. */
 	if (!ret)
-		lrng_es_add_entropy();
+		lrng_force_fully_seeded();
 
 out:
 	mutex_unlock(&lrng_crypto_cb_update);
@@ -275,7 +275,7 @@ int lrng_set_hash_cb(const struct lrng_hash_cb *hash_cb)
 	 * any waiting entropy providers.
 	 */
 	if (!ret) {
-		lrng_es_add_entropy();
+		lrng_force_fully_seeded();
 		lrng_writer_wakeup();
 	}
 
-- 
2.39.2.501.gd9d677b2d8

